component/azstorage/datalake.go

```diff
@@ -553,7 +553,40 @@ func (dl *Datalake) ReadInBuffer(name string, offset int64, len int64, data []by
 
 // WriteFromFile : Upload local file to file
 func (dl *Datalake) WriteFromFile(name string, metadata map[string]*string, fi *os.File) (err error) {
-	return dl.BlockBlob.WriteFromFile(name, metadata, fi)
+	// File in DataLake may have permissions and ACL set. Just uploading the file will override them.
+	// So, we need to get the existing permissions and ACL and set them back after uploading the file.
+
+	var acl string = ""
```

Comment:
Refactor the function 
and move setACL getACL in proper file


func (dl *Datalake) WriteFromFile(name string, metadata map[string]*string, fi *os.File) (err error) {
	var acl string
	var fileClient *file.Client

	// Only retrieve ACL if we need to preserve it
	if dl.Config.preserveACL {
		// Prepare the file client for the specific file
		fileClient = dl.Filesystem.NewFileClient(filepath.Join(dl.Config.prefixPath, name))

		// Get the current ACL
		if acl, err = getACL(fileClient); err != nil {
			log.Err("Datalake::getACL : Failed to get ACLs for file %s [%s]", name, err.Error())
		}
	}

	// Upload the file; this can overwrite any existing ACLs/permissions
	if err = dl.BlockBlob.WriteFromFile(name, metadata, fi); err != nil {
		return fmt.Errorf("failed to upload file: %v", err)
	}

	// If ACL was retrieved, restore it after the upload
	if acl != "" {
		if err = setACL(fileClient, acl); err != nil {
			log.Err("Datalake::WriteFromFile : Failed to set ACL for %s [%s]", name, err.Error())
		}
	}

	return nil
}

// Helper function to retrieve ACL
func getACL(fileClient *file.Client) (string, error) {
	resp, err := fileClient.GetAccessControl(context.Background(), nil)
	return *resp.ACL, err
}

// Helper function to set ACL
func setACL(fileClient *file.Client, acl string) error {
	_, err := fileClient.SetAccessControl(context.Background(), &file.SetAccessControlOptions{
		ACL: &acl,
	})
	return err
}

Reply:
getACL is not used anywhere else hence did not create it a seperate method. Can do if that improves code readabiltiy.